Framework/PythonInterface/mantid/api/src/Exports/ExperimentInfo.cpp

```diff
@@ -81,7 +81,8 @@ void export_ExperimentInfo() {
 
       .def("detectorInfo", &ExperimentInfo::detectorInfo,
```

Comment:
What happens if this method is called and `ExperimentInfo::setInstrument` has not been run? This would happen if you were to create a workspace with no instrument, for example using `CreateWorkspace` to create the workspace.

Reply:
Ah I see. I think a runtime error would occur? Could a possible solution be to have an intermediate method which checks if the instrument has been set and if not it could throw an error?

Reply:
@OwenArnold All workspaces have an instrument, it will simply have 0 detectors, i.e., the size of `DetectorInfo` is empty. I do not see an issue here. Am I overlooking something?